PlusServerLauncher/PlusServerLauncherMainWindow.cxx

```diff
@@ -890,3 +963,169 @@ void PlusServerLauncherMainWindow::OnTimerTimeout()
+
+  vtkSmartPointer<vtkXMLDataElement> commandElement = vtkSmartPointer<vtkXMLDataElement>::New();
+  commandElement->SetName("Command");
+  vtkSmartPointer<vtkXMLDataElement> startServerElement = vtkSmartPointer<vtkXMLDataElement>::New();
+  startServerElement->SetName("ServerStarted");
+  startServerElement->SetAttribute("LogLevel", logLevelString.c_str());
+  startServerElement->SetAttribute("ConfigFileName", filename.c_str());
+
+  vtkSmartPointer<vtkXMLDataElement> configFileElement = NULL;
+  configFileElement = vtkSmartPointer<vtkXMLDataElement>::Take(vtkXMLUtilities::ReadElementFromFile(configFilePath));
+  if (configFileElement)
+  {
+    std::string ports = this->GetServersFromConfigFile(filename);
+    startServerElement->SetAttribute("Servers", ports.c_str());
+    startServerElement->AddNestedElement(configFileElement);
+  }
+  commandElement->AddNestedElement(startServerElement);
+
+  igtl::MessageBase::MetaDataMap map;
+  map["LogLevel"] = std::pair<IANA_ENCODING_TYPE, std::string>(IANA_TYPE_US_ASCII, startServerElement->GetAttribute("LogLevel"));
+  map["ConfigFileName"] = std::pair<IANA_ENCODING_TYPE, std::string>(IANA_TYPE_US_ASCII, startServerElement->GetAttribute("ConfigFileName"));
+  if (startServerElement->GetAttribute("Servers"))
+  {
+    map["Servers"] = std::pair<IANA_ENCODING_TYPE, std::string>(IANA_TYPE_US_ASCII, startServerElement->GetAttribute("Servers"));
+  }
+
+  std::stringstream commandStream;
+  vtkXMLUtilities::FlattenElement(commandElement, commandStream);
+  this->SendCommand(PlusServerLauncherMainWindow::PLUS_SERVER_LAUNCHER_REMOTE_DEVICE_ID, "ServerStarted", commandStream.str(), map);
+
+}
+
+//---------------------------------------------------------------------------
+void PlusServerLauncherMainWindow::SendServerStoppedCommand(const char* configFilePath)
+{
+  LOG_TRACE("Sending server stopped command");
+
+  vtkSmartPointer<vtkXMLDataElement> commandElement = vtkSmartPointer<vtkXMLDataElement>::New();
+  commandElement->SetName("Command");
+
+  igtl::MessageBase::MetaDataMap map;
+  map["ConfigFileName"] = std::pair<IANA_ENCODING_TYPE, std::string>(IANA_TYPE_US_ASCII, configFilePath);
+
+  std::stringstream commandStream;
+  vtkXMLUtilities::FlattenElement(commandElement, commandStream);
+  this->SendCommand(PlusServerLauncherMainWindow::PLUS_SERVER_LAUNCHER_REMOTE_DEVICE_ID, "ServerStopped", commandStream.str(), map);
+}
+
+//---------------------------------------------------------------------------
+void PlusServerLauncherMainWindow::OnLogEvent(vtkObject* caller, unsigned long eventId, void* clientData, void* callData)
+{
+  PlusServerLauncherMainWindow* self = reinterpret_cast<PlusServerLauncherMainWindow*>(clientData);
+
+  if (!self->m_RemoteControlServerConnector || !self->m_RemoteControlServerConnector->IsConnected())
+  {
+    return;
+  }
+
+  const char* rawLogMessage = static_cast<char*>(callData);

```

Comment:
This isn't a valid assumption. The logging system supports logging of wchar as well.